src/backend/distributed/commands/alter_table.c

```diff
@@ -1122,9 +1125,27 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 									quote_qualified_identifier(schemaName, sourceName))));
 		}
 
-		appendStringInfo(query, "INSERT INTO %s SELECT * FROM %s",
-						 quote_qualified_identifier(schemaName, targetName),
-						 quote_qualified_identifier(schemaName, sourceName));
+		if (HasAnyGeneratedStoredColumns(sourceId))
+		{
+			/* see comment in BuildInsertColumnStringForReplaceTable */
```

Comment:
I'd move that comment to here

Reply:
addressed all three of them in 7fa29c15b19672cb9119802068d50e21387b9f04